use binary search to get seq from offset

staltz opened the pull request:
## Context

I'm preparing for #199 and one thing I realized is that I'll need a quick way of getting the seq from any given offset, e.g. during compaction there is going to be `unshiftedOffset` and I'd like to discover `unshiftedSeq`.

## Problem

We don't have a helper function for that and currently we're just doing a linear for-loop to look for the offset.

Another problem is that the "end" of the for-loop is `tarr.length`, which I think is wrong because the tarr grows much more than the `count` does (like it doubles in size every time it "grows" but most of it is still unused). Instead we need to use `index.count`.

## Solution

Introduce `getSeqFromOffset` that does a binary search and uses `indexes['seq'].count` as the end marker.

This should also have a performance improvement for those cases when the jitdb indexes are many and large but we just need to update the tip of the index. This way we can avoid scanning the whole index in `O(n)` and can do `O(log n)`. I'm not sure we have any benchmark for this but should be a pretty common thing to happen in production.

staltz commented:
Here's previous benchmarks to compare with: https://github.com/ssb-ngi-pointer/jitdb/pull/197

But there's no noticeable difference. But do we have a benchmark to measure this case? (A big index receives a bit of updates)

arj03 commented:
This looks like a great optimization. I'm wondering if we can't we use [bsb](https://www.npmjs.com/package/binary-search-bounds) that is already included?

staltz commented:
Good point! I switched to using bsb.eq in getSeqFromOffset, and I took a look at the source code, it's basically the same, so that's good.

I also added another commit [use seq index.count instead of index.tarr.length](https://github.com/ssb-ngi-pointer/jitdb/pull/200/commits/bcba09eff01981fa8e384883adfafcdbbbd6580f).

arj03 reviewed (approved):
Excellent. This is a good improvement.
